prow/plugins/bugzilla/bugzilla.go

```diff
@@ -356,12 +356,13 @@ To reference a bug, add 'Bug XXX:' to the title of this pull request and request
 			log.Debug("Valid bug found.")
 			response = fmt.Sprintf(`This pull request references `+bugLink+`, which is valid.`, e.bugId, bc.Endpoint(), e.bugId)
 			// if configured, move the bug to the new state
-			if options.StatusAfterValidation != nil && bug.Status != *options.StatusAfterValidation {
-				if err := bc.UpdateBug(e.bugId, bugzilla.BugUpdate{Status: *options.StatusAfterValidation}); err != nil {
+			update := options.StateAfterValidation.AsBugUpdate()
+			if update != nil && (update.Status != bug.Status || update.Resolution != bug.Resolution) {
+				if err := bc.UpdateBug(e.bugId, *update); err != nil {
 					log.WithError(err).Warn("Unexpected error updating Bugzilla bug.")
-					return comment(formatError(fmt.Sprintf("updating to the %s state", *options.StatusAfterValidation), bc.Endpoint(), e.bugId, err))
+					return comment(formatError(fmt.Sprintf("updating to the %s state", prettyState(*options.StateAfterValidation)), bc.Endpoint(), e.bugId, err))
```

Comment:
If we make the bug type a `Stringer` by implementing `.String()` we just `%s` and are done

Reply:
I did not know `Stringer`!